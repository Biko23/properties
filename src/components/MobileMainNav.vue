<template>
  <v-list nav dense style="margin-left: 10px">
    <v-list-group :value="false">
      <template v-slot:activator>
        <v-list-item-title>Buy</v-list-item-title>
      </template>
      <v-list>
        <links title="Properties for sale" link="/properties-for-sale" />
        <!-- <links title="New Constructions" />
                    <links title="For sale by owner" />
                    <links link="/" title="Open houses" /> -->
      </v-list>
    </v-list-group>

    <v-list-group :value="false">
      <template v-slot:activator>
        <v-list-item-title>Sell</v-list-item-title>
      </template>
      <v-list>
        <links title="Sell A Property" link="/property-requirement" />
        <!-- <links title="Current market trends" /> -->
        <links title="Seller's guide" link="/learn" />
        <links title="Property Details" link="/property-details" />
        <!-- <links title="Price Conversion Calculator" /> -->
      </v-list>
    </v-list-group>

    <v-list-group :value="false">
      <template v-slot:activator>
        <v-list-item-title>Rent</v-list-item-title>
      </template>
      <v-list>
        <links link="/property" title="Houses for Rent" />
        <links link="/register" title="List A Rental" />
        <!-- <links link="/login" title="Apartments for Rent" />
                        <links link="/description" title="Your Rentals" /> -->
      </v-list>
    </v-list-group>

    <v-list-group :value="false">
      <template v-slot:activator>
        <v-list-item-title>Home Loan</v-list-item-title>
      </template>
      <v-list>
        <links title="Mortgage Lenders" link="/mortgage" />
        <links title="Mortgage Rates" link="/mortgage-rates" />
        <!-- <links title="Mortgage Refinancing" />
                    <links link="/" title="Mortgage Calculator" /> -->
      </v-list>
    </v-list-group>

    <v-list-group :value="false">
      <template v-slot:activator>
        <v-list-item-title>Find Service Provider</v-list-item-title>
      </template>
      <v-list>
        <template v-for="category in allVendorsCategories">
          <links
            :key="category.value"
            :link="`/provider/${category.value}`"
            :title="category.text"
          />
        </template>
      </v-list>
    </v-list-group>
    <v-list-group :value="false">
      <template v-slot:activator>
        <v-list-item-title>Get Started</v-list-item-title>
      </template>
      <v-list>
        <links title="Join Us As?" link="/getstarted" />
      </v-list>
    </v-list-group>
    <!-- <v-col class="hid-navbar">
        <v-btn
          color="primary"
          style="text-decoration: none"
          v-if="loginState"
          @click="logingOut"
          >Logout</v-btn
        >
        <router-link to="/login" style="text-decoration: none" v-else
          >Login</router-link
        >
      </v-col> -->
  </v-list>
</template>
<script>
import { mapActions, mapGetters } from "vuex";
import Links from "@/components/Links";
export default {
  name: "MobileMainNav",
  components: {
    Links,
  },
  data: () => ({
    vendorCategories: [],
  }),
  computed: {
    ...mapGetters(["loginState", "allVendorsCategories"]),
  },
  created() {
    this.fetchVendorsCategories();
  },
  methods: {
    ...mapActions(["logout", "fetchVendorsCategories"]),
    async logingOut() {
      try {
        await this.logout().then(() => {
          this.$router.push("/");
        });
      } catch (error) {
        throw new Error(error);
      }
    },
  },
};
</script>