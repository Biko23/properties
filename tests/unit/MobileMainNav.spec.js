import { createLocalVue, shallowMount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import VueRouter from 'vue-router'
import Vuex from 'vuex'
import MobileMainNav from '@/components/MobileMainNav.vue'

describe('MobileMainNav.vue', () => {
    const localVue = createLocalVue()
    localVue.use(Vuex)
    localVue.use(VueRouter)
    let vuetify
    let actions
    let getters
    let store
    let router

    beforeEach(() => {
        vuetify = new Vuetify()
        router = new VueRouter()
        actions = {
            logout: jest.fn(),
            fetchVendorsCategories: jest.fn()
        },
        getters = {
            loginState: jest.fn(),
            allVendorsCategories: jest.fn(),
            iAmACertifiedSeller: jest.fn()
        }
        store = new Vuex.Store({
            actions,
            getters
        })
    })

    const mountFunction = options => {
        return shallowMount(MobileMainNav, {
            localVue,
            vuetify,
            store,
            router,
            ...options
        })
    }

    it('should have a match snapshot', () => {
        const wrapper = mountFunction()
        expect(wrapper.html()).toMatchSnapshot()
    })
})

