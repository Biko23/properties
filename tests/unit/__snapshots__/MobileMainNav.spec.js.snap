// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileMainNav.vue should have a match snapshot 1`] = `
<v-list-stub tag="div" dense="true" nav="true" style="margin-left: 10px;">
  <v-list-group-stub color="primary" activeclass="" appendicon="$expand" ripple="true">
    <v-list-stub tag="div">
      <links-stub link="/properties-for-sale" title="Properties for sale"></links-stub>
    </v-list-stub>
  </v-list-group-stub>
  <v-list-group-stub color="primary" activeclass="" appendicon="$expand" ripple="true">
    <v-list-stub tag="div">
      <links-stub link="/property-requirement" title="Sell A Property"></links-stub>
      <links-stub link="/learn" title="Seller's guide"></links-stub>
      <!---->
    </v-list-stub>
  </v-list-group-stub>
  <v-list-group-stub color="primary" activeclass="" appendicon="$expand" ripple="true">
    <v-list-stub tag="div">
      <links-stub link="/properties-for-rent" title="Houses for Rent"></links-stub>
      <links-stub link="/rental-requirement" title="List A Rental"></links-stub>
    </v-list-stub>
  </v-list-group-stub>
  <v-list-group-stub color="primary" activeclass="" appendicon="$expand" ripple="true">
    <v-list-stub tag="div">
      <links-stub link="/mortgage" title="Mortgage Lenders"></links-stub>
      <links-stub link="/mortgage-rates" title="Mortgage Rates"></links-stub>
    </v-list-stub>
  </v-list-group-stub>
  <v-list-group-stub color="primary" activeclass="" appendicon="$expand" ripple="true">
    <v-list-stub tag="div"></v-list-stub>
  </v-list-group-stub>
  <v-list-group-stub color="primary" activeclass="" appendicon="$expand" ripple="true">
    <v-list-stub tag="div">
      <links-stub link="/getstarted" title="Join Us As?"></links-stub>
    </v-list-stub>
  </v-list-group-stub>
  <v-col-stub tag="div" class="hid-navbar">
    <router-link-stub to="/login" tag="a" ariacurrentvalue="page" event="click" style="text-decoration: none;">Login</router-link-stub> <span style="margin-left: 15px;">OR</span>
    <router-link-stub to="/signup" tag="a" ariacurrentvalue="page" event="click" style="text-decoration: none; margin-left: 15px;">Sign up</router-link-stub>
  </v-col-stub>
</v-list-stub>
`;
